src/sockperf.cpp

```diff
@@ -1718,7 +1718,16 @@ static int resolve_sockaddr(const char *host, const char *port, int sock_type,
     int res = getaddrinfo(host, port, &hints, &result);
     if (res == 0) {
         get_socket_address(result, addr, addr_len);
+    } else if (res == EAI_NONAME && host != NULL) { // Unix Domain Socket name
+        addr_len = strlen(host) + 2;
+        printf("host is %s\n", host);
+        strncpy(((sockaddr_store_t*)addr)->addr_un.sun_path, host, addr_len);
+        ((sockaddr_store_t*)addr)->addr_un.sun_family = AF_UNIX;
+        ((sockaddr_store_t*)addr)->ss_family = AF_UNIX;
+
+        return 0;
```

Comment:
```suggestion
        res = 0;
```
otherwise you will get memory leak at line 1731 (freeaddrinfo)

Reply:
Will be fixed.

Reply:
@agalanin-at-nvidia 
I am not sure it's correct because getaddrinfo was failed and the error that was returned is EAI_NONAME. 
using freeaddrinfo in that case will cause segmentation fault (attaching screenshot) 


![segmentation fault for freeaddrinfo](https://user-images.githubusercontent.com/44922030/164436342-a4acc3b5-cb71-439d-8d96-8c7ffedecc01.JPG)

Reply:
ok. please move freeaddinfo into `if (res == 0)` branch

Reply:
created separate PR